daml-assistant/src/DAML/Assistant/Cache.hs

```diff
@@ -24,13 +26,29 @@ newtype CacheKey = CacheKey String
 newtype CacheTimeout = CacheTimeout NominalDiffTime
     deriving (Show, Eq, Ord, Y.FromJSON)
 
+data UpdateCheck
+    = UpdateCheckNever
+    | UpdateCheckEvery CacheTimeout
+    deriving (Show, Eq, Ord)
+
+instance Y.FromJSON UpdateCheck where
+    parseJSON (Y.String "never") = pure UpdateCheckNever
+    parseJSON y = UpdateCheckEvery <$> Y.parseJSON y
+
 cacheLatestSdkVersion
     :: DamlPath
     -> IO (Maybe SdkVersion)
     -> IO (Maybe SdkVersion)
-cacheLatestSdkVersion =
-    cacheWith "latest-sdk-version" (CacheTimeout 86400)
-        serializeMaybeSdkVersion deserializeMaybeSdkVersion
+cacheLatestSdkVersion damlPath getVersion = do
+    damlConfigE <- try $ readDamlConfig damlPath
+    let updateCheckM = join $ eitherToMaybe (queryDamlConfig ["update-check"] =<< damlConfigE)
+        defaultUpdateCheck = UpdateCheckEvery (CacheTimeout 86400)
+    case fromMaybe defaultUpdateCheck updateCheckM of
+        UpdateCheckNever -> pure Nothing
```

Comment:
Is this right? I would have expected their to be an explicit `daml update` command (because it's super handy to say check now regardless), and I would expect if someone set UpdateNever then did `daml update` (or merely had ever been on UpdateCheckEvery beforehand) it would next time return the most recent version it had cached, but not check afresh.

Reply:
That makes sense. Does `daml update` also install the latest version if it finds a new one?

Reply:
Maybe? Not sure. Are you doing to suggest `daml install` already does what I want?

Reply:
`daml install latest --activate` will try to install the latest version of the sdk and assistant, regardless of the `update-check` setting.